c10/core/SymInt.h

```diff
@@ -47,12 +48,46 @@ class C10_API SymInt {
     return data_ == p2.data_;
   }
 
+  bool operator!=(const SymInt& p2) const {
+    return data_ != p2.data_;
+  }
+
   SymInt operator+(SymInt sci) const {
     TORCH_CHECK(
         !this->is_symbolic() && !sci.is_symbolic(),
         "Symbolic Add isn't supported yet");
     return SymInt(data_ + sci.data_);
   }
+  bool operator<(SymInt sci) const {
+    TORCH_CHECK(
+        !this->is_symbolic() && !sci.is_symbolic(),
+        "Symbolic lt isn't supported yet");
+    return data_ < sci.data_;
+  }
+  void operator*=(SymInt sci) {
+    TORCH_CHECK(
+        !this->is_symbolic() && !sci.is_symbolic(),
+        "Symbolic mul_ isn't supported yet");
+    data_ = data_ * sci.data_;
+  }
+
+  bool operator<(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic lt isn't supported yet");
+    return data_ < sci;
+  }
+  bool operator==(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic eq isn't supported yet");
+    return data_ == sci;
+  }
+  bool operator!=(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic neq isn't supported yet");
+    return data_ != sci;
+  }
+
+  SymInt operator*(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic mul isn't supported yet");
+    return SymInt(data_ * sci);
+  }
 
   std::shared_ptr<SymbolicIntNode> toSymbolicIntNode();
   static c10::SymInt toSymInt(std::shared_ptr<SymbolicIntNode> sin);
```

Comment:
re data below, it would be nice to not directly use data(), but name it something scary like unchecked_expect_int() to note that there is a precondition that it must not be symbolic. I wish I could dump data too but IValue would not work.